src/components/KCatalog/KCatalog.vue

```diff
@@ -177,366 +177,347 @@
-     */
-    errorStateActionRoute: {
-      type: [Object, String],
-      default: '',
-    },
-    /**
-     * A prop to pass in a custom error state action message
-     */
-    errorStateActionMessage: {
-      type: String,
-      default: '',
-    },
-    /**
-     * A prop to pass in a custom error state icon
-     */
-    errorStateIcon: {
-      type: String,
-      default: '',
-    },
-    /**
-     * A prop to pass in a color for the error state icon
-     */
-    errorStateIconColor: {
-      type: String,
-      default: '',
-    },
-    /**
-     * A prop to pass in a size for the error state icon
-     */
-    errorStateIconSize: {
-      type: String,
-      default: '50',
-    },
-    /**
-     * A prop to pass in a fetcher function to enable server-side pagination
-     */
-    fetcher: {
-      type: Function,
-      required: true,
-    },
-    /**
-     * A prop to pass in a an object of intial params for the initial fetcher function call
-     */
-    initialFetcherParams: {
-      type: Object,
-      default: null,
-    },
-    /**
-     * A prop to trigger a revalidate of the fetcher function. Modifying this value
-     * will trigger a manual refetch of the table data.
-     */
-    fetcherCacheKey: {
-      type: String,
-      default: '',
-    },
-    /**
-     * A prop to pass in a search string for server-side search
-     */
-    searchInput: {
-      type: String,
-      default: '',
-    },
-    /**
-     * A prop to pass in a the number of pagination neighbors used by the pagination component
-     */
-    paginationNeighbors: {
-      type: Number,
-      default: 1,
-    },
-    /**
-     * A prop to pass in an array of page sizes used by the pagination component
-     */
-    paginationPageSizes: {
-      type: Array as PropType<number[]>,
-      default: () => ([15, 30, 50, 75, 100]),
-      validator: (pageSizes: number[]): boolean => !!pageSizes.length && pageSizes.every(i => typeof i === 'number'),
-    },
-    /**
-     * A prop to pass the total number of items in the set for the pagination text
-     */
-    paginationTotalItems: {
-      type: Number,
-      default: null,
-    },
-    disablePaginationPageJump: {
-      type: Boolean,
-      default: false,
-    },
-    disablePagination: {
-      type: Boolean,
-      default: false,
-    },
-    /**
-     * A prop to pass to hide pagination for total table records is less than or equal to pagesize
-     */
-    hidePaginationWhenOptional: {
-      type: Boolean,
-      default: false,
-    },
-    /**
-     * for testing only, strips out generated ids and avoid loading state in tests.
-     * 'true' - no id's no loading
-     * 'loading' - no id's but allow loading
-     */
-    testMode: {
-      type: [String, Boolean],
-      default: false,
+  cardSize: {
+    type: String as PropType<CardSize>,
+    default: 'medium',
+    validator: (value: CardSize): boolean => {
+      return Object.values(cardSizeRecord).includes(value)
```

Comment:
**nitpick**: another way to handle types and validators
```
export const CardSizeArray = ['size 1', 'size 2'] as const

export type CardSize = AnyElementOf<typeof CardSizeArray>
```

`validator: (value: CardSize): boolean => CardSizeArray.includes(value)`

Reply:
Feedback applied 👍🏻 . I also pushed up `KCatalogItem` converted to setup syntax (missed it in my first pass).